maintenance/wikia/MassSetWikiFactoryValues.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+require_once __DIR__ . '/../Maintenance.php';
+
+# format of the data should follow the 
+# https://docs.google.com/spreadsheets/d/1vGuJEZ0ncVQSXTrABM3YYHg2P531sWZosgNsq-uMy5M/edit#gid=2073314697
+
+class MassSetWikiFactoryValues extends Maintenance {
+	protected $saveChanges  = false;
+	
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = 'Sets Wiki Factory variable values based on CSV';
+		$this->addArg( 'file', 'CSV file with the list of wikis and vars' );
+		$this->addOption( 'saveChanges', 'Change the wiki values for real.', false, false, 'd' );
+	}
+	
+	private function readFromCSV() {
+		$fileName = $this->getArg( 0 );
+		
+		$fileHandle = fopen( $fileName, 'r' );
```

Comment:
Let's validate the fileHandle and exit if it's bad.

Reply:
d3a5b5e